src/django_elasticsearch_dsl_drf/viewsets.py

```diff
@@ -39,7 +40,9 @@ def __init__(self, *args, **kwargs):
         self.client = connections.get_connection()
         self.index = self.document._doc_type.index
         self.mapping = self.document._doc_type.mapping.properties.name
-        self.search = Search(using=self.client, index=self.index)
+        doc_type = self.document.__name__
+        doc_type = re.sub(r'(.)([A-Z])', r'\1_\2', doc_type).lower()
+        self.search = Search(using=self.client, index=self.index, doc_type=doc_type)
```

Comment:
The `doc_type` value might be taken directly from `self.document._doc_type.name`. No need for fuzzy `re` guesses.
So, I think only the following line might stay:

```python
self.search = Search(
    using=self.client, 
    index=self.index, 
    doc_type=self.document._doc_type.name
)
```

Please, make changes and update the pull request.

Reply:
sounds cool i will update that.. let me check

Reply:
yeah. made the change, works like a charm

Reply:
@rrmerugu:

Good. Thanks.